### ISSUE TITLE
Allow multiple plots from one plot file

### ISSUE THREAD

**Issue by mattlbeck** [iterative/dvc#6316]
Consider a mock dvc.yaml such as:

```
stages:
    A:
        cmd: ...
        plots:
            - metrics.csv:
                x: x
                y: y1
            - metrics.csv:
                x: x
                y: y2
```

DVC will only show one (the last) plot. The fix is to split into multiple metrics filer with the same x but different y variables. This feels pretty inconvenient for metrics that are related in their dependent variable, and so can be stored in the same table and file.

On the surface I don't see any reason why a specified plot artifacts can only be used for one plot.

**Comment by pared** [comment]
Related #5980

**Comment by shcheklein** [comment]
Relevant to iterative/studio-support#23.

**Comment by shcheklein** [comment]
@dberenbaum should we/can we prioritize this?

**Comment by dberenbaum** [comment]
See a proposed spec for this in https://github.com/iterative/dvc/discussions/5980#discussioncomment-1026072.

@shcheklein Yup, we will be working on it soon. @pared What do you think? I think we had planned to address #6603 first to get image support looking a little better, but this is probably more important given the number of requests for it.

**Comment by daavoo** [comment]
By "combining" we have been usually referring to get multiple (`y`) columns from either the same file or multiple files, which seems reasonable given how we usually recommend formatting the data in linear plots.

However, I think that the user in https://github.com/iterative/studio-support/issues/23#issuecomment-929932069 showcases a new scenario that we haven't discussed so far in https://github.com/iterative/dvc/discussions/5980#discussioncomment-1026072 . It is basically using the values of a third column (`z`) to group the values of a single column (`y`) and the template looks very clean.

IMO this use case could be easily covered on the DVC side by extending the vega template and adding a new placeholder
`DVC_METRIC_COLOR_LABEL` and we could encourage this format as a quick workaround for combining `y` values. On the DVCLive side, we could add some logic to facilitate the optional usage of this format (i.e. adding a new `stage` column for grouping based on the `metric` name).

**Comment by daavoo** [comment]
I think that #6603 still needs to be addressed first because the `diff` operation for this kind of combined plots would require implementing some sort of facet grid in the HTML layout (looks like Studio would need to implement something similar as well), as @dberenbaum mentioned in https://github.com/iterative/dvc/discussions/5980#discussioncomment-1008543 .

**Comment by dberenbaum** [comment]
> However, I think that the user in [iterative/studio-support#23 (comment)](https://github.com/iterative/studio-support/issues/23#issuecomment-929932069) showcases a new scenario that we haven't discussed so far in [#5980 (reply in thread)](https://github.com/iterative/dvc/discussions/5980#discussioncomment-1026072) . It is basically using the values of a third column (`z`) to group the values of a single column (`y`) and the template looks very clean.

I think this could even be a separate issue to add grouping by a categorical column. We could add a template like `linear_categories` and add the `DVC_METRIC_COLOR_LABEL`. We could also add a similar template for the scatter plot (and maybe others like bar plots in the future).

**Comment by pared** [comment]
While grouping by categorical values sounds like a valid use case, I would not focus on it right away. My reasoning for this is the user had to come up with a workaround for the currently not existing feature. This needs to be addressed but I think we should not focus on it because it is about adjusting existing plots implementation to unsupported use case. I think we should focus on #5980 and come up with a process that will allow us to handle that. Main research should focus on two things:
1. How to structure `Repo.plots` data so that we can meaningfully group both images and vega datapoints.
2. How to implement rendering for Vega - ideally we should limit number of plots to as small as possible, if its possible to do "multiple grouping" ex `revision` -> `filename/stage_name` we might not require new template for that.

**Comment by dberenbaum** [comment]
I agree that it seems better to support multi-series plots natively in a more generic way. Users should not have to create custom templates or adjust their data format to make it work. The train/test example is often going to have data in different files, and we shouldn't ask users to combine them to plot them together. However, we could point users to that workaround in the meantime.

@pared What about handling the proposal in https://github.com/iterative/dvc/discussions/5980#discussioncomment-1026072 and parsing data from that syntax? Do we need more research there?

**Comment by pared** [comment]
Yes, we need to define how `diff` will behave, and consider how more advanced use cases (like parallel coordinates) plots will be defined using the proposed approach.

**Comment by dberenbaum** [comment]
I think it would be great if parallel coordinates plots could be made via `dvc plots`, but I think there are alternatives. If it's only part of `dvc exp show --html` or similar, it doesn't necessarily need to fit cleanly into the `dvc plots` syntax, right?

**Comment by pared** [comment]
@dberenbaum at this time it does not correspond at all, I just need to make a POC and research how hard it will be to generalize its behavior.

**Comment by pared** [comment]
But besides that I believe we can start implementation right away.

**Comment by pared** [comment]
@mattlbeck The implementation should be on master.

for the use case from the description the plot would look like as follows:

```yaml
stages:
  ...

plots:
  metrics.csv:
    x: x
    y: [y1, y2]
    
# alternatively:
  my_plot:
    x: x
    y: 
      metrics.csv: [y1, y2]
```
